tensorboard/components/tf_scalar_dashboard/tf-scalar-chart.html

```diff
@@ -0,0 +1,316 @@
+    `color` can be set to display a colored border at the left of the
+        card; if left unset, no border will be displayed:
+    Any contents of the heading will be rendered below the title, and
+        can be used to display a subtitle, some small control widgets,
+        or similar.
+-->
+<dom-module id="tf-card-heading">
+  <template>
+    <div class="title-container" style="border-color: [[_borderColor]]">
+      <div
+        class="title"
+        inner-h-t-m-l="[[_break(title)]]"></div>
+      <div class="content">
+        <content></content>
+      </div>
+    </div>
+    <style>
+      .title-container {
+        border-left: 4px solid;
+        padding-left: 5px;
+        margin-bottom: 10px;
+      }
+      .title {
+        font-size: 14px;
+        text-overflow: ellipsis;
+        overflow: hidden;
+      }
+      .content {
+        font-size: 12px;
+      }
+    </style>
+  </template>
+  <script>
+    Polymer({
+      is: "tf-card-heading",
+      properties: {
+        title: String,
+        color: {
+          type: String,
+          value: null,   // this property is optional
+        },
+        _borderColor: {
+          type: String,
+          computed: '_computeBorderColor(color)',
+          readOnly: true,
+        },
+      },
+      _computeBorderColor(color) {
+        return color || 'rgba(255, 255, 255, 0.0)';  // 100% transparent white
+      },
+      _break: function(input) {
+        return input.replace(/([\/_-])/g, "$1<wbr>")
+      },
+    });
+  </script>
+</dom-module>
+
+<!--
+  A component that fetches scalar data from the TensorBoard server and
+  renders it into a vz-line-chart.
+-->
+<dom-module id="tf-scalar-chart">
+  <template>
+    <!--
+      Uncomment these to test the behavior of tf-card-heading.
+      They render like this:
+          card expanded:  http://i.imgur.com/Q7lbvxk.png
+          not expanded:   http://i.imgur.com/M38yshU.png
+      (In the "not expanded" one, the vz-line-chart is way too small,
+      but that's just because the headings take up too much space. It
+      doesn't indicate any problems with the headings.)
+    -->
+    <!--
+    <tf-card-heading title="[[tag]]" color="red">
+      I'm a subtitle, for testing.
+    </tf-card-heading>
+    <tf-card-heading title="[[tag]]" color="blue">
+      Look, I'm like the image dashboard. It's also fine if my content becomes really exorbitantly wide. Nothing to see here.<br />
+      step <strong>500</strong><br />
+      <input type="range" style="width: 100%"></input>
+    </tf-card-heading>
+    -->
+    <tf-card-heading title="[[tag]]"></tf-card-heading>
+    <vz-line-chart
+      x-type="[[xType]]"
+      color-scale="[[_runsColorScale]]"
+      smoothing-enabled="[[smoothingEnabled]]"
+      smoothing-weight="[[smoothingWeight]]"
+      tooltip-sorting-method="[[tooltipSortingMethod]]"
+      ignore-y-outliers="[[ignoreYOutliers]]"
+    ></vz-line-chart>
+    <div style="display: flex; flex-direction: row;">
```

Comment:
Yes: simplicity. CSS classes are often redundant in the context of WebComponents/React components. The purpose of CSS classes in vanilla HTML is to let you style many different instances of one component in the same way: all my `div class="toggle-log-scale-container"`s can behave in the same way. But here we only have one instance: the _component_ has multiple instances.